enhancements/microshift/microshift-tls-cipher.md

```diff
@@ -0,0 +1,233 @@
+
+For simplicity, the API server cipher suites will also apply to internal
+control plane components:
+* API server
+* Kubelet
+* Kube controller manager
+* Kube scheduler
+* Etcd
+* Route controller manager
+
+Except for API server the rest of the components are internal to MicroShift
+and are not exposed through an endpoint but require secure connections too.
+
+IMPORTANT: Because of limitations in the Go implementation of TLS, using
+version 1.3 does not allow to change the cipher suites. Whenever TLS 1.3 is
+configured in `tls.minVersion`, `tls.cipherSuites` will take the default suites
+for TLS 1.3, ignoring and overwriting whatever was configured.
+
+### Workflow Description
+**cluster admin** is a human user responsible for configuring a MicroShift
+cluster.
+
+1. The cluster admin adds specific configuration for API server cipher suites
+   prior to MicroShift's start.
+2. After MicroShift starts, all the control plane components (API server, kube
+   controller manager, kube scheduler) will be configured with the specified
+   cipher suites.
+3. All clients connecting to API server must support the configured cipher
+   suites or else connections will fail in TLS handshake phase.
+
+### API Extensions
+As described in hte proposal section, there is one new configuration option:
+```yaml
+apiServer:
+  tls:
+    cipherSuites: # Defaults to the suites of the configured minVersion
+    - <cipher suite 1>
+    - ...
+    minVersion: <VersionTLS12|VersionTLS13> # Defaults to 1.2
+```
+
+### Topology Considerations
+#### Hypershift / Hosted Control Planes
+N/A
+
+#### Standalone Clusters
+N/A
+
+#### Single-node Deployments or MicroShift
+Enhancement is solely intended for MicroShift.
+
+### Implementation Details/Notes/Constraints
+As mentioned in the proposal, there is one limitation in regards to TLS 1.3
+that relates to Golang implementation.
+
+[This issue](https://github.com/golang/go/issues/29349) brings up the question
+about configuring them and the release notes for Golang state:
+```
+TLS 1.3 cipher suites are not configurable. All supported cipher suites are
+safe, and if PreferServerCipherSuites is set in Config the preference order is
+based on the available hardware.
+```
+
+So all of the settings for MicroShift about cipher suites will only apply to
+TLS 1.2.
+
+Practically speaking, this means any attempt to configure `tls.cipherSuites`
+when having `tls.minVersion` set to 1.3 will result in MicroShift ignoring and
+overwriting the contents to match the cipher suites that Golang will use. Note
+that this is only for informational purposes, no cipher suites will be passed
+on to the http servers, as they are ignored.
+
+### Risks and Mitigations
+N/A
+
+### Drawbacks
+In the event of having a compromised cipher suite in TLS 1.3 an admin would not
+be able to configure the API server to not use it.
+Golang restricts configuration in this regard, so the fix should come from the
+language libraries themselves.
```

Comment:
Just out of curiosity: is it something shipped with the language (stdlib) or it's a package fetched from the internet? In other words, if that ever happens, will it: rebuild with newer Go or be bump in go.mod?

Reply:
This is from go tls library itself: https://cs.opensource.google/go/go/+/refs/tags/go1.23.4:src/crypto/tls/common.go;l=676